app/services/github_issues_author.rb

```diff
@@ -0,0 +1,55 @@
+class GithubIssuesAuthor
+  attr_reader :existing_authors_ids
+
+  ALLOWED_ACTIONS = %w[opened edited deleted].freeze
+
+  def initialize(author_ids = [])
+    @existing_authors_ids = author_ids
+  end
+
+  def perform(action, issue)
+    return unless ALLOWED_ACTIONS.include?(action) && valid_issue?(issue)
+
+    send("issue_#{action}", issue)
+  end
+
+  private
+
+  def valid_issue?(issue)
+    %w[id title body].all? { |attribute| issue.key? attribute }
+  end
+
+  def issue_opened(issue)
+    return if existing_authors_ids.include?(issue['id'])
+
+    ActiveRecord::Base.transaction do
```

Comment:
I'm using a transaction here to make sure i don't have any dummy data either it's going to be created entirely or nothing at all